aiohttp/http_exceptions.py

```diff
@@ -86,8 +86,8 @@ def __init__(
 class InvalidHeader(BadHttpMessage):
     def __init__(self, hdr: Union[bytes, str]) -> None:
         if isinstance(hdr, bytes):
-            hdr = hdr.decode("utf-8", "surrogateescape")
-        super().__init__(f"Invalid HTTP Header: {hdr}")
+            hdr = repr(hdr)
+        super().__init__(f"Invalid HTTP header: {hdr}")
         self.hdr = hdr
```

Comment:
Would it make sense to use `backslashreplace`? as `repr()` is going to result in a str that looks like `"b'foo'"`. That probably also breaks backwards-compatibility on the hdr attribute.

It might be safer to use repr() in the main message though (where there's a minor security issue as someone could try and manipulate the message to look like it's not part of the header).

Maybe something like this?
```
hdr_s = hdr.decode(errors="backslashreplace") if isinstance(hdr, bytes) else hdr
super().__init__(f"Invalid HTTP header: {hdr!r}")
self.hdr = hdr_s
```